packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -61,523 +62,621 @@ const _ = require('lodash');
-          .observe(resolvedGas);
-    };
-
-    /**
-     * Internal helper method that removes the leading 0x if there is one.
-     * @param input
-     * @private
-     */
-    private static prune0x(input: string): string {
-        return input.startsWith('0x')
-            ? input.substring(2)
-            : input;
+      if (e instanceof JsonRpcError) {
+        throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
+      }
+      throw sdkClientError;
     }
-
-    private static HbarToWeiBar(balance: AccountBalance): BigNumber {
-        return balance.hbars
-        .to(HbarUnit.Tinybar)
-        .multipliedBy(constants.TINYBAR_TO_WEIBAR_COEF);
+  }
+
+  private captureMetrics = (mode, type, status, cost, gas, caller, interactingEntity) => {
+    const resolvedCost = cost ? cost : 0;
+    const resolvedGas = typeof gas === 'object' ? gas.toInt() : 0;
+    this.consensusNodeClientHistogramCost.labels(mode, type, status, caller, interactingEntity).observe(resolvedCost);
+    this.consensusNodeClientHistogramGasFee.labels(mode, type, status, caller, interactingEntity).observe(resolvedGas);
+  };
+
+  /**
+   * Internal helper method that removes the leading 0x if there is one.
+   * @param input
+   * @private
+   */
+  private static prune0x(input: string): string {
+    return input.startsWith('0x') ? input.substring(2) : input;
+  }
+
+  private static HbarToWeiBar(balance: AccountBalance): BigNumber {
+    return balance.hbars.to(HbarUnit.Tinybar).multipliedBy(constants.TINYBAR_TO_WEIBAR_COEF);
+  }
+
+  private createFile = async (
+    callData: Uint8Array,
+    client: Client,
+    requestId?: string,
+    callerName?: string,
+    interactingEntity?: string,
+  ) => {
+    const requestIdPrefix = formatRequestIdMessage(requestId);
+    const hexedCallData = Buffer.from(callData).toString('hex');
+
+    const fileCreateTx = await new FileCreateTransaction()
+      .setContents(hexedCallData.substring(0, 4096))
+      .setKeys(client.operatorPublicKey ? [client.operatorPublicKey] : []);
+    const fileCreateTxResponse = await fileCreateTx.execute(client);
+    const { fileId } = await fileCreateTxResponse.getReceipt(client);
+
+    const createFileRecord = await fileCreateTxResponse.getRecord(this.clientMain);
+    this.captureMetrics(
+      SDKClient.transactionMode,
+      fileCreateTx.constructor.name,
+      Status.Success,
+      createFileRecord.transactionFee.toTinybars().toNumber(),
+      createFileRecord?.contractFunctionResult?.gasUsed,
+      callerName,
+      interactingEntity,
+    );
+
+    if (fileId && callData.length > 4096) {
+      const fileAppendTx = await new FileAppendTransaction()
+        .setFileId(fileId)
+        .setContents(hexedCallData.substring(4096, hexedCallData.length))
+        .setChunkSize(4096)
+        .setMaxChunks(this.maxChunks);
+      const fileAppendTxResponse = await fileAppendTx.execute(client);
+
+      const fileAppendRecord = await fileAppendTxResponse.getRecord(this.clientMain);
```

Comment:
I don't think this is sufficient.
If you see from https://github.com/hashgraph/hedera-sdk-js/blob/develop/src/file/FileAppendTransaction.js#L367 the SDK submits multiple individual transactions, however it only returns the 1st.
As such you'll inly get the response of the 1st.

I think we have at least 2 options
1. Perform the separate FileAppends ourselves
2. Query all the transaction records based on chunk size. Each transaction will be separated by a non second so you have predict al the transactionIds

Reply:
Got it, fixed.